server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -2176,6 +2193,8 @@ public void onFailure(Exception e) {
 
             @Override
             public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
+                listener.onResponse(newDelete);
```

Comment:
I moved this right up here in the interest of having it complete as early as possible, this means we resolve this listener before failing the snapshot completion listeners below. Happy to move it back if that's a significant re-ordering of events.

Reply:
Yeah I think I'd prefer to leave this alone. Or at least make this change in a separate PR so we can bisect back to it if it causes problems. But really I'd prefer to leave it.